packages/rspack/src/ModuleFederationPlugin.ts

```diff
@@ -75,6 +75,28 @@ export class ModuleFederationPlugin implements RspackPluginInstance {
       this._patchChunkSplit(compiler, options.name);
     }
 
+    if (options.experiments?.provideExternalRuntime) {
+      if (options.exposes) {
+        throw new Error(
```

Comment:
why can we only do this to pure consumers?

Reply:
I think in the future we should be able to support both. 
in my previous PR i had setup something that allowed the remote of a build to use a different federation entry file which used the external referenced package. So i think its possible to do the same here but i will adjust my PR

Reply:
to avoid duplicate inject runtime , if it has exposes , it's usually will be consumed